test/test-index.ts

```diff
@@ -102,21 +90,40 @@ test('should return only the first error even if multiple would throw', (t) => {
       el.setAttribute('rendered', '');
     },
     'amp-fail1': (_el: Element) => {
-      throw new Error('Cannot render <amp-fail1>');
+      throw new Error('Something is undefined');
     },
     'amp-fail2': (_el: Element) => {
-      throw new Error('Cannot render <amp-fail2>');
+      throw new Error('Something is undefined');
     },
   };
 
   const ast = treeProto(
     h('amp-success', {}, [h('amp-fail1'), h('amp-fail1'), h('amp-fail2')])
   );
-  const expected: Result<string> = {
-    type: 'failure',
-    error: new Map([['amp-fail1', ['Cannot render <amp-fail1>']]]),
+  t.throws(() => renderAst(ast, instructions), {message: /amp-fail/});
+});
+
+test('should allow a custom error handler', (t) => {
+  const instructions = {
+    'amp-success': (el: Element) => {
+      el.setAttribute('rendered', '');
+    },
+    'amp-fail1': (_el: Element) => {
+      throw new Error('Something is undefined');
+    },
+    'amp-fail2': (_el: Element) => {
+      throw new Error('Something is undefined');
+    },
   };
-  t.deepEqual(renderAst(ast, instructions), expected);
+
+  const ast = treeProto(
+    h('amp-success', {}, [h('amp-fail1'), h('amp-fail1'), h('amp-fail2')])
+  );
+
+  const errors = [];
+  const handleError = (tagName) => errors.push(tagName);
```

Comment:
Ignore me
```suggestion
  const handleError = errors.push.bind(errors);
```

Reply:
I'm relatively anti-bind, so will ignore this one (but I like all your other thoughts!)